UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs

```diff
@@ -32,6 +32,18 @@ public static void PlaySoundAndShake(Vector3Int worldPosition, byte shakeIntensi
 
 			_ = SoundManager.PlayNetworkedAtPosAsync(explosionSound, worldPosition, audioSourceParameters, true, false, shakeParameters);
 
+			if (distantSound?.AudioSource.OrNull()?.maxDistance == null)
+			{
+				Logger.LogError($"distantSound?.AudioSource.OrNull()?.maxDistance == null From {distantSound}" );
+				return;
+			}
+
+			if (groanSound?.AudioSource.OrNull()?.maxDistance == null)
```

Comment:
this on appears to hide the problem rather than fix it, can you look into why it's returning null when the audio source and it's data are already there?

Reply:
while I was testing C4's I also noticed that this only happens on `groanSound` not `distantSound`

Reply:
Might be a null value in CommonSounds

Reply:
shall I fix the missing reference? but I don't know what the sound should be?

Reply:
it's not hiding the problem it's just catching it and logging